app/src/spIP/java/org/dhis2/usescases/sms/DI/SmsModule.kt

```diff
@@ -0,0 +1,81 @@
+package org.dhis2.usescases.sms.DI
+
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import org.dhis2.usescases.sms.data.api.ConstantApi
+import org.dhis2.usescases.sms.data.api.ConstantApiImpl
+import org.dhis2.usescases.sms.data.api.OutboundApi
+import org.dhis2.usescases.sms.data.api.OutboundApiImpl
+import org.dhis2.usescases.sms.data.repository.message.MessageTemplateD2Repository
+import org.dhis2.usescases.sms.data.repository.patient.PatientD2Repository
+import org.dhis2.usescases.sms.data.repository.preferred.PreferredLanguageD2Repository
+import org.dhis2.usescases.sms.data.repository.sms.SmsApiRepository
+import org.dhis2.usescases.sms.domain.repository.message.MessageTemplateRepository
+import org.dhis2.usescases.sms.domain.repository.patient.PatientRepository
+import org.dhis2.usescases.sms.domain.repository.preferred.PreferredLanguageRepository
+import org.dhis2.usescases.sms.domain.repository.sms.SmsRepository
+import org.dhis2.usescases.teiDashboard.TeiDashboardMenuCustomActionsManager
+import org.dhis2.usescases.teiDashboard.ui.TeiDashboardMenuCustomActionsManagerImpl
+import org.hisp.dhis.android.core.D2Manager
+import org.hisp.dhis.android.core.arch.api.HttpServiceClient
+
+@Module
```

Comment:
Use simple service locator like in 2.9.1 to avoid conflict problems and to have to add sms code in other flavors